### DISCUSSION TITLE
V8Runtime heap size limit tuning

### DISCUSSION THREAD

**Discussion by Lookout-ent** [caoccao/Javet#292] [Q&A]
i was wondering if there is any to set the V8Runtime heapSizeLimit ?

**Comment by caoccao** [comment]
Yes, almost all [V8 options](https://docs.google.com/document/d/1DFsbyoPcdK1__fe1nBDz3Vtr50DCa8W2aj5I-dFpPlg) are supported.

Please check [this file](https://github.com/caoccao/Javet/blob/main/src/test/java/com/caoccao/javet/BaseTestJavet.java) out.

```java
setMaxHeapSize(...);
setMaxOldSpaceSize(...);
setCustomFlags("--max-semi-space-size=... --min-semi-space-size=...");
```

Please be careful. The V8 flags must be set during application initialization. Once the first V8 runtime is created, the V8 flags are sealed and further modification to the V8 flags will no longer take effect.

**Reply by Lookout-ent** [reply]
In V8 runtime i'm getting OOM issue , while in Node.js runtime i'm not seeing any out of memory issue and script execution is passing.
Any suggestions ?

I'm using Javet-2.2.2

How i'm using javet ?  ---
I'm creating a pool of engines of V8 runtime type.
Each engine is initialized with the same script.

I'm using v8Runtime.getV8HeapStatistics() to examine the heap stats for that engine
After initializing the engine with script -- used heap memory is 4532776
After invoking my function --- heap size limit is 4345298944 and used heap size is 2321441336
are these numbers in bytes ?!

**Reply by Lookout-ent** [reply]
I want to use V8 runtime only ... but not able to understand how do i exactly tune the heap size limit 
heap size limit in V8 mode is 1518338048(1.5GB) , while in Node.js mode, it is 4GB
Since the test passes in Node.js mode , i assume i need to tune the heap size limit in V8 mode
I check V8Flags  but not able to understand how to use this while initializing the engine pool 

```
@Override
    protected JavetEngine createEngine() throws JavetException {
        //JavetEngine engine = super.createEngine();
        JavetEngineHelper engine = new JavetEngineHelper(this, super.createEngine().getV8Runtime(), scriptContent);
        engine.getV8Runtime().getExecutor(scriptContent).executeVoid();
        return engine;
    }
```
createEngine does not V8RuntimeOptions as an argument

any help is appreciated 
thanks

**Reply by caoccao** [reply]
V8 options are set during the V8 initialization. It's a global, one-time-only setting. So, it's a static field in Javet. The sample file in my last reply has that part. Please let me know if you have any questions.

**Reply by Lookout-ent** [reply]
thanks ... got it
